### PR TITLE
Update to php8 and SF6

### PR DESCRIPTION
**Pull request by kdurillon**
### Compatibilité PHP8 et SF6

> Suppression du atoum-bundle qui n'est pas dispo en php8

> Correction des deprecated

> MAJ de php cs-fixer

### CODE REVIEW

File: src/Tests/Units/Redis/RedisClient.php
Comment (DocRoms) [inline src/Tests/Units/Redis/RedisClient.php:52]: 🎖️

File: src/Tests/Units/Redis/RedisSessionHandler.php
Comment (DocRoms) [inline src/Tests/Units/Redis/RedisSessionHandler.php:15]: 🏅 👍

File: src/Tests/Units/DependencyInjection/M6WebRedisExtension.php
Comment (DocRoms) [inline src/Tests/Units/DependencyInjection/M6WebRedisExtension.php:14]: 🏅 👍

File: src/Tests/Units/AbstractTest.php
Comment (DocRoms) [inline src/Tests/Units/AbstractTest.php:7]: Bien vu ! 👍

**Review by DocRoms — APPROVED** [review]

File: composer.json
Comment (FloFrad) [inline composer.json:16]: Is it possible to keep the php7 compatibility?
```suggestion
        "php": "^7.4|>=8.0",
```

Comment (kdurillon) [reply composer.json:16]: I'm using union types to correct deprecations in RedisSessionHandler and it's a php 8 feature 😕

Comment (FloFrad) [reply composer.json:16]: You code like a pig 🐷 😜

**Review by FloFrad — APPROVED** [review]

File: src/Redis/RedisSessionHandler.php
Comment (Fabex) [inline src/Redis/RedisSessionHandler.php:58]: Pourrais tu supprimer ce else qui est vraiment inutile stp ;)
ca me désespère de voir des "if return else return"

Comment (kdurillon) [reply src/Redis/RedisSessionHandler.php]: done :)

**Comment by kdurillon** [comment]
@FloFrad @Fabex Si c'est ok pour vous est ce que l'un de vous peut merge/créer un tag ? ça nous débloquerait pour la migration de notre côté 🙂🙏

File: composer.json
Comment (shavounet) [inline composer.json:26]: This bundle uses a bunch of symfony component, why isn't it in `require` ? (see #63, maybe there's a good reason, I just don't know :grin:  )

**Review by shavounet — APPROVED** [review]

**Comment by kdurillon** [comment]
Updated the PR with @shavounet comments

File: composer.json
Comment (shavounet) [inline composer.json:20]: I don't think we should require the whole symfony framework, but instead only required components.
```suggestion
        "symfony/cache": "~5.0 || ~6.0",
        "symfony/config": "~5.0 || ~6.0",
        "symfony/dependency-injection": "~5.0 || ~6.0",
        "symfony/event-dispatcher-contracts": "~5.0 || ~6.0",
        "symfony/http-foundation": "~5.0 || ~6.0",
        "symfony/http-kernel": "~5.0 || ~6.0"
```

There is also the case of `m6web/guzzle-http-bundle` but maybe let's not make it mandatory but only in `suggest` ? I don't really know if it can have any impact being missing, or being required...

Comment (kdurillon) [reply composer.json]: We do not require m6web/guzzle-http-bundle in our project and got this bundle to work fine so it's definitely not mandatory

File: composer.json
Comment (Oliboy50) [inline composer.json:25]: if we support `5.0` then we also support `4.4`

```suggestion
        "symfony/cache": "~4.4 || ~5.0 || ~6.0",
        "symfony/config": "~4.4 || ~5.0 || ~6.0",
        "symfony/dependency-injection": "~4.4 || ~5.0 || ~6.0",
        "symfony/event-dispatcher-contracts": "~4.4 || ~5.0 || ~6.0",
        "symfony/http-foundation": "~4.4 || ~5.0 || ~6.0",
        "symfony/http-kernel": "~4.4 || ~5.0 || ~6.0"
```

File: composer.json
Comment (shavounet) [reply composer.json]: Yeah I suppose your're right, it's only an opt-in class

**Review by Oliboy50 — COMMENTED** [review]
thanks but our CI does not start on your PR, I will try to fix this

**Comment by Oliboy50** [comment]
@kdurillon I made [this PR](https://github.com/BedrockStreaming/RedisBundle/pull/65) to fix our CI

I'm sorry but you will have to wait for it to be merged, then you'll have to rebase your PR on master, and update the CI script to work on php 8.0 and php 8.1 🙏

**Review by Fabex — APPROVED** [review]

**Comment by Fabex** [comment]
Quand tu auras bien tout rebase par rapport à master ca sera ok pour merge et release une nouvelle version majeur

**Comment by kdurillon** [comment]
i had to drop SF4 compatibility as it is not available fro PHP8

**Comment by kdurillon** [comment]
I had to change min versions of predis, atoum and cs-fixer to ensure the CI passed with prefer lowest on 8.1.
Should be all good now

File: .php-cs-fixer.php
Comment (Oliboy50) [inline .php-cs-fixer.php:1]: please remove this file, I already configured php-cs-fixer in the previous PR

File: composer.json
Comment (Oliboy50) [inline composer.json:28]:
```suggestion
        "m6web/php-cs-fixer-config": "2.0.*",
```

(we don't want to use the `2.1.0` version in this project... at least not in this PR)

File: composer.json
Comment (Oliboy50) [inline composer.json:25]: please support `~4.4`

File: composer.json
Comment (Oliboy50) [inline composer.json:33]:
```suggestion
```

Comment (kdurillon) [reply composer.json:33]: Min version for php 8.1, see this failed ci : https://github.com/BedrockStreaming/RedisBundle/actions/runs/2232153155

Comment (Oliboy50) [reply composer.json:33]: ok, we probably have to fix this directly in `m6web/php-cs-fixer-config` then

it is fine to merge as is and remove it later when this will be fixed 👌

**Review by Oliboy50 — APPROVED** [review]
thanks :)

**Comment by Oliboy50** [comment]
I don't know much about the predis API but my mates approved and I'm sure you know what you're doing, so I'll merge your PR, thanks again

**Pull request merged**
